src/compiler/crystal/semantic/cleanup_transformer.cr

```diff
@@ -306,6 +306,45 @@ module Crystal
       node
     end
 
+    def transform(node : MultiAssign)
+      if @program.has_flag?("preview_multi_assign")
```

Comment:
The flag name `preview_multi_assign` should be reconsidered with a fresh perspective. I guess initially it was chosen from a slightly different context -- both of the mentioned related changes -- but now it's just one.

The reason I think it may be confusing now is that "preview" sounds like it's gonna give you a good new thing, but in reality it only gives you more errors, so a "bad" thing when directly contrasted with the other expectation.

I would like the flag to include the word "strict". Then if the word is simply added in, the flag name may sound too long. I don't know if we should then try to eliminate some other word from it or rethink it somehow.
One word I thought about removing was "preview", but that becomes a general decision: shall we always include the word "preview" in opt-in flags for future features, or not.

Reply:
> shall we always include the word "preview" in opt-in flags for future features, or not.

In my understanding, a preview is optional/alternative behaviour that's available to try out and experiment, but it's not entirely finished. In contrast, opt-in features are complete and expected to become the default in a subsequent breaking release. Obviously, there's some grey area, but I think that distinction makes sense. `preview_mt` is a preview and we are still not sure how and when it becomes a stable feature.

For this PR, I think we're mostly in agreement that the strict behaviour shall be adopted and become the new default in 2.0.

Reply:
So my suggestion for the flag name would be `strict_multi_assign`.